app/Services/Backup/ShellProcessor.php

```
@@ -30,17 +31,21 @@ public function process(string $command): string
-            $incrementalOutput .= $data;
-
-            // Update the log entry with incremental output (sanitized)
-            if ($this->logger && $logIndex !== null) {
-                $this->logger->updateCommandLog($logIndex, [
-                    'output' => $this->sanitize(trim($incrementalOutput)),
-                    'duration_ms' => round((microtime(true) - $startTime) * 1000, 2),
-                ]);
-            }
-        });
+        try {
+            $process->run(function ($type, $data) use (&$incrementalOutput, $logIndex, $startTime) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$type'. (undefined)

(UnusedFormalParameter)